### DISCUSSION TITLE
Updated Rapsodo Range creates a flicker after each shot and is consistently causing ghost shots

### DISCUSSION THREAD

**Discussion by bkauhl3** [springbok/MLM2PRO-GSPro-Connector#61] [General]
I just got my first sim set up and working well with Awesome Golf. I have been watching a lot of Joe's videos and I decided to try GSPro following his instructions. I am not sure if this is a known issue or if I have something set up incorrectly, but my Rapsodo Range doesn't look like the one in his videos. I am assuming it is just the new updated one. Anyway, I set up the OCR areas in the new places (at the bottom of the screen) and it was sending all of the correct data. But, after each shot the rapsodo range appeared to flash/flicker on my ipad, causing a ghost shot in GSPro. I am not sure if this is an iPad issue or if anyone else is experiencing this same thing after the Rapsodo Range update. Thanks for any help and thanks for making this connector!

**Comment by springbok** [comment]
Have set the range camera to stationary, if you haven’t you will get ghost shots.

**Reply by bkauhl3** [reply]
Yes, I had it set to stationary. Like the whole screen flickers once really fast after each shot and it picks up the numbers again.

**Reply by bkauhl3** [reply]
I'm just wondering if anyone else has had that happen or if I need to try with a different iPad. I believe my current iPad is from 2021

**Comment by springbok** [comment]
So the iPad screen flickers? 

The connector takes a screenshot every 0.25 seconds and if the screen changes and the numbers are then interpreted differently by the OCR it could generate ghost shots. So if it is flickering then it may cause issues.

**Comment by bkauhl3** [comment]
Here's a video of the flicker. It seems like it's happening in the Rapsodo app and after your last shot stops rolling.        https://github.com/springbok/MLM2PRO-GSPro-Connector/assets/33236808/b423abc3-9123-4616-be9e-4965d8276e1b

**Reply by wonder99** [reply]
I see the red line flicker, but not the numbers? Do you see numbers flicker?

**Reply by bkauhl3** [reply]
I don't see the numbers flicker either. Also, fwiw, when casting to my projector the tracer doesn't flash on the big screen. It only flashes on the iPad. Here are a few more shots with my lights turned off so the screen is more clear. 

Would the connector be able to ignore a shot if all of the numbers are the exact same as the previous shot? 

https://github.com/springbok/MLM2PRO-GSPro-Connector/assets/33236808/94c8fe0a-d07f-41ad-b07f-2ccdc57ca38a

**Reply by wonder99** [reply]
Yes, the connector already does that.  Can you send you log file from the appdata logs together?  Delete the file first, then start it up and take a couple of shots. The zip up the file and attach it here

**Reply by bkauhl3** [reply]
[mlm2pro-gspro-connect.zip](https://github.com/springbok/MLM2PRO-GSPro-Connector/files/13824800/mlm2pro-gspro-connect.zip)

It was less frequent tonight. I don't know if it has anything to do with the fact that there were 2 players the other night and only 1 tonight. Regardless, the final shot I took was followed by a ghost shot. There were a couple others throughout the round but I'm not sure how obvious they will be to find. Thanks for taking the time to check this out.

**Reply by bkauhl3** [reply]
After hitting more shots on the range, I am noticing that there are a few misreads (30 degree VLA on Rapsodo vs. 3 degrees on GSPro) that are usually followed by a ghost shot. Like the OCR is misinterpreting the numbers initially and then reading them either correctly or differently a second time. If you have any suggestions on modifying how i have set up my ROI's that may be a place I can check. I had to fiddle with them quite a bit before i could verify all of them. I checked that they are all correct but I will mess with the VLA and see if I can get better results.

**Comment by springbok** [comment]
Can you try this version and see if it improved the misreads. Can you also send a screenshot of your ROI’s.

https://1drv.ms/u/s!Agx7aYfjrZJ8oiugn4k0_wU7Zhvj?e=WVa4G3

**Reply by jmjava** [reply]
i have the new version ( V1.01.23 ) will start testing when my net is delivered
![image](https://github.com/springbok/MLM2PRO-GSPro-Connector/assets/591751/58eefd73-8d51-4203-b6a9-3e9ffada69a3)

**Comment by jmjava** [comment]
Getting a net next week, if this is still going on I may be able to finally try to help.  I have IPad with Mlm2pro and Exputt but waiting on the Net.

**Comment by bkauhl3** [comment]
The other day I was able to resolve most of the OCR misreads by setting the Graphics Quality to High on the Rapsodo Range. I am not sure if that was already mentioned somewhere but maybe it will help someone else in the future. I have hit a few hundred shots since then and it has been pretty consistent. Thanks to everyone for assisting me through this!

**Comment by jmjava** [comment]
lol my android is on its way :)  if nothing else, it will enable me to test
both I guess

Great find.  You should send a note to Joe lagowski - he has been reporting
on this
